src/content/reference/react/use-server.md

```diff
@@ -24,34 +24,233 @@ canary: true
+* Directives like `'use server'` must be at the very beginning of their function or file, above any other code including imports (comments above directives are OK). They must be written with single or double quotes, not backticks.
 
 ## Usage {/*usage*/}
 
-<Wip>
-This section is a work in progress. 
+### Server actions in forms {/*server-actions-in-forms*/}
+
+The most common use case of server actions will be calling server functions that mutate data. On the browser, the [HTML form element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) is the traditional approach for a user to submit a mutation. With React Server Components, React introduces first-class support for server actions in [forms](/reference/react-dom/components/form).
+
+Here is a form that allows a user to request a username.
+
+```js
+// UsernameForm.js
+'use client';
+
+export default UsernameForm({submit}) {
+  return (
+    <form action={submit}>
+      <input type="text" name="username" />
+      <button type="submit">Request</button>
+    </form>
+  );
+}
+```
+
+```js
+// App.js
+import UsernameForm from './UsernameForm';
+
+function requestUsername(formData) {
+  'use server';
+  const username = formData.get('username');
+  // ...
+}
+
+export default App() {
+  <UsernameForm submit={requestUsername}>
+}
+```
+
+In this example `UsernameForm` is a Client Component and receives a server action as the prop `submit`. When a user submits this form, there is a network request to the server function `requestUsername`.
+
+When calling a server action in a `<form action>`, React will supply the form's [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) as the first argument to the server action.
+
+<Note>
+
+We force `UsernameForm` as a Client Component with `'use client'` purely for demonstration. In general, it is not recommended to add the `'use client'` directive if the component does not use [client-only features](reference/react/use-client#usage).
+
+</Note>
+
+#### Progressive Enhancement in `<form>` {/*progressive-enhancement-in-form*/}
+
+In the above example, the `<form>` is rendered in a Client Component. By instead rendering the `<form>` in a Server Component, React can support [progressively enhanced](https://developer.mozilla.org/en-US/docs/Glossary/Progressive_Enhancement) forms.
+
+In this context, we mean that server-rendered forms can provide the baseline experience of submitting a form without JavaScript. This is useful for apps that are server-side rendered (SSR) where HTML scaffolding is sent to the client first, before any code is downloaded to add interactivity.
+
+<DeepDive>
+
+#### What is the difference between SSR and React Server Components (RSC)? {/*what-is-the-difference-between-ssr-and-react-server-components-rsc*/}
```

Comment:
is this the right page for this explanation? doesn't seem actions-specific

Reply:
Yea I wasn't sure about this, it needs to go somewhere as I just feel like if I read this I'd be confused. I guess we can punt it to learn docs.